crates/view/src/storage.rs

```diff
@@ -75,20 +77,37 @@ pub struct Storage {
 
 impl Storage {
     /// If the database at `storage_path` exists, [`Self::load`] it, otherwise, [`Self::initialize`] it.
+    #[tracing::instrument(
+        skip_all,
+        fields(
+            path = ?storage_path.as_ref().map(|p| p.as_ref().as_str()),
+            url = %node,
+        )
+    )]
     pub async fn load_or_initialize(
         storage_path: Option<impl AsRef<Utf8Path>>,
         fvk: &FullViewingKey,
         node: Url,
     ) -> anyhow::Result<Self> {
-        if let Some(path) = storage_path.as_ref() {
-            if path.as_ref().exists() {
+        if let Some(path) = storage_path.as_ref().map(AsRef::as_ref) {
+            if path.exists() {
+                tracing::debug!(?path, "database exists");
                 return Self::load(path).await;
+            } else {
+                tracing::debug!(?path, "database does not exist");
             }
         };
 
-        let mut client = AppQueryServiceClient::connect(node.to_string()).await?;
+        let mut client = AppQueryServiceClient::connect(node.to_string())
+            .instrument(error_span!("connecting to endpoint"))
+            .await
+            .tap_err(|error| {
+                tracing::error!(?error, "failed to connect to app query service endpoint")
+            })?
+            .tap(|_| tracing::debug!("connected to app query service endpoint"));
         let params = client
             .app_parameters(tonic::Request::new(AppParametersRequest {}))
+            .instrument(error_span!("getting app parameters"))
```

Comment:
ditto nit

Reply:
addressed in 6189d376fd157e1fe06b106fcbd089318470aea9 :hammer: